tests/task_critical_region_test.toit

```diff
@@ -29,3 +30,14 @@ test_cancel_can_timeout:
       with_timeout --ms=1:
         critical_do:
           sleep --ms=10000
+
+test_deadline_in_critical:
+  with_timeout --ms=100:
+    expect_not_null task.deadline
+    critical_do:
+      expect_not_null task.deadline
+
+  with_timeout --ms=100:
+    expect_not_null task.deadline
+    critical_do --no-respect_deadline:
+      expect_null task.deadline
```

Comment:
Maybe add a test that shows that the deadline doesn't trigger
```
  expect_no_throw:
    with_timeout --ms=10:
      critical_do:
        sleep --ms=50
```

Reply:
Ah, you mean from within the critical region. I will try that.

Reply:
Done.